components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -80,6 +85,72 @@
 
+    private String stsRedirectPage = null;
+
+    /**
+     * This method reads Passive STS Html Redirect file content.
+     * This should have been implemented in the backend but done in the front end to avoid API changes
+     *
+     * @return Passive STS Html Redirect Page File Content
+     */
+    private String readPassiveSTSHtmlRedirectPage() {
+        String redirectHtmlFilePath = CarbonUtils.getCarbonHome() + File.separator + "repository"
+                + File.separator + "resources" + File.separator + "security" + File.separator +
+                "sts_redirect.html";
+        FileInputStream fileInputStream = null;
+        String fileContent = null;
+        try {
+            fileInputStream = new FileInputStream(new File(redirectHtmlFilePath));
+            fileContent = new Scanner(fileInputStream, "UTF-8").useDelimiter("\\A").next();
+
+            if (log.isDebugEnabled()) {
+                log.debug("sts_redirect.html : " + fileContent);
+            }
+
+        } catch (FileNotFoundException e) {
+            // The Passive STS Redirect HTML file is optional. When the file is not found, use the default page content.
+            log.info("Passive STS Redirect HTML file not found in : " + redirectHtmlFilePath +
+                    ". Default Redirect is used.");
+        } finally {
+            if (fileInputStream != null) {
+                try {
+                    fileInputStream.close();
+                } catch (IOException e) {
+                    log.error("Error occurred when closing file input stream for sts_redirect.html", e);
+                }
+            }
+        }
+
+        if (fileContent == null || fileContent.trim().length() == 0) {
+            fileContent = "<html>" +
+                    "    <body>" +
+                    "        <p>You are now redirected to $url ." +
+                    "           If the redirection fails, please click the post button." +
+                    "        </p>" +
+                    "        <form method='post' action='$url'>" +
+                    "        <p>" +
+                    "            <!--$params-->" +
+                    "            <!--$additionalParams-->" +
+                    "            <button type='submit'>POST</button>" +
+                    "       </p>" +
+                    "       </form>" +
+                    "        <script type='text/javascript'>" +
+                    "            document.forms[0].submit();" +
+                    "        </script>" +
+                    "    </body>" +
+                    "</html>";
+        }
+
+        // Adding parameters to the Passive STS HTML redirect page
+        String parameters = "<input type='hidden' name='wa' value='$action'>" +
+                "<input type='hidden' name='wresult' value='$result'>" +
+                "<input type='hidden' name='wctx' value='$context'>";
+
+        fileContent = fileContent.replace("<!--$params-->", parameters);
```

Comment:
Since you are replacing the "<!--$params-->" string, add a check for the existance of this string if the content are read from file